### ISSUE TITLE
Many <Mention> bug

### ISSUE THREAD

**Issue by sofianegargouri** [effektif/react-mentions#143]
Hi guys ! :)

I have an issue with the following snippet:

    <MentionsInput value={this.props.value} className="textarea"
                              onChange={this.handleChange.bind(this)}
                              markup="[__display__](__type__:__id__)"
                              displayTransform={this.transformDisplay}>
        <Mention trigger="#"
                     type="tag"
                     data={this.state.tags}
                     renderSuggestion={this.renderTagSuggestion}
                     style={{color: '#0bc4e0', background: '#fff'}}/>

        <Mention trigger="@"
                     type="user"
                     data={this.state.mentions}
                     renderSuggestion={this.renderUserSuggestion}
                     style={{color: '#b0b0b0', background: '#fff'}}/>
    </MentionsInput>

Here is how it isn't working:
When I type # in my textarea first, I have the good suggestions (the tags), but then, if I want to mention a user by typing @, it's loading my tags AND the users.
If I type @ first and try to # then, I have the same problem, but with the users loaded first.

Thanks ! 😄

**Comment by hirish** [comment]
I have similar code:
```
<MentionsInput
    style={style}
    value={value}
    onKeyDown={this.onKeyDown}
    placeholder={placeholder}
    onChange={this.onChange}
    markup="@{__type__|__id__|__display__}"
    ref="input"
>

    <Mention
        type="USER"
        trigger="@"
        className={STYLES.mention}
        renderSuggestion={renderSuggestion}
        data={users.toJS()}
    />

    <Mention
        type="PROCESS"
        trigger="#"
        className={STYLES.mention}
        renderSuggestion={renderSuggestion}
        data={searchProcesses}
    />

</MentionsInput>
```

This works in 0.6.2, appears to have broken in 0.6.3.

**Comment by hirish** [comment]
Presumably as a result of https://github.com/effektif/react-mentions/pull/142

**Comment by jpstrikesback** [comment]
Seeing this as well.

**Comment by jfschwarz** [comment]
Fixed in v0.6.4 with #148. Thank you, @mersocarlin! 👌
